ccm_web/client/src/components/CreateSectionWidget.tsx

```diff
@@ -0,0 +1,101 @@
+import { CanvasCourseSection } from '../models/canvas'
+import { CCMComponentProps } from '../models/FeatureUIData'
+import { CanvaCoursesSectionNameValidator, ICanvasSectionNameInvalidError } from '../utils/canvasSectionNameValidator'
+import { CODE_NUMPAD_ENTER, CODE_RETURN } from 'keycode-js'
+
+const useStyles = makeStyles((theme) => ({
+  root: {
+    backgroundColor: '#FAFAFA',
+    height: 200
+  },
+  input: {
+    width: '100%',
+    paddingRight: '2px'
+  },
+  button: {
+    width: '100%'
+  }
+}))
+
+export interface CreateSectionWidgetProps extends CCMComponentProps {
+  onSectionCreated: (newSection: CanvasCourseSection) => void
+}
+
+function CreateSectionWidget (props: CreateSectionWidgetProps): JSX.Element {
+  const classes = useStyles()
+  const { enqueueSnackbar } = useSnackbar()
+  const [newSectionName, setNewSectionName] = useState<string>('')
+  const [isCreating, setIsCreating] = useState(false)
+  const nameValidator = new CanvaCoursesSectionNameValidator(props.globals.course)
+
+  const newSectionNameChanged = (event: ChangeEvent<HTMLInputElement>): void => {
+    setNewSectionName(event.target.value)
+  }
+
+  const errorAlert = (errors: ICanvasSectionNameInvalidError[]): void => {
+    const errorMessage = errors.map(e => { return e.reason }).join('<br/>')
+    enqueueSnackbar(errorMessage, {
+      variant: 'error'
+    })
+  }
+
+  const createSection = (): void => {
+    if (newSectionName.trim().length === 0) {
+      return
+    }
+    setIsCreating(true)
+    nameValidator.validateSectionName(newSectionName).then(errors => {
+      if (errors.length === 0) {
+        addCourseSections(props.globals.course.id, [newSectionName])
+          .then(newSections => {
+            props.onSectionCreated(newSections[0])
+            setNewSectionName('')
+          }).catch(() => {
+            enqueueSnackbar('Error adding section', {
+              variant: 'error'
+            })
+          })
+      } else {
+        errorAlert(errors)
+      }
+    }).catch(() => {
+      enqueueSnackbar('Error validating section name', {
+        variant: 'error'
+      })
+    }).finally(() => {
+      setIsCreating(false)
+    })
+  }
+
+  const isCreateDisabled = (): boolean => {
+    return isCreating || newSectionName.trim().length === 0
+  }
+
+  const keyDown = (code: string): void => {
+    if (isCreateDisabled()) return
+    if (code === CODE_RETURN || code === CODE_NUMPAD_ENTER) {
+      createSection()
+    }
+  }
+
+  return (
+    <>
+    <Grid container>
+      <Grid item xs={9}>
+        <TextField className={classes.input} size='small' label='New Section Name' variant='outlined' id="outlined-basic" onChange={newSectionNameChanged} value={newSectionName} onKeyDown={(e) => keyDown(e.code)}/>
+      </Grid>
+      <Grid item>
+        <Button className={classes.button} variant="contained" color="primary" onClick={createSection} value={newSectionName} disabled={isCreateDisabled()}>
+          Create
+        </Button>
```

Comment:
**Current State from PR**
![Screen Shot 2021-08-30 at 2 37 00 PM](https://user-images.githubusercontent.com/8579775/131388022-9c61cdd0-31e5-433d-adf1-6310569b3a36.png)



**UI Wireframes recommendation:**
![Screen Shot 2021-08-30 at 2 36 09 PM](https://user-images.githubusercontent.com/8579775/131387973-67509be9-03da-471c-b22e-e9ee1cf1cb7f.png)
![Screen Shot 2021-08-30 at 2 32 49 PM](https://user-images.githubusercontent.com/8579775/131388493-ac4ee6ee-bf32-405c-b713-bbf3bcd6257f.png)

I see number of minor discrepancies for create section UI from the Mockup. I am going to list them below 
1. The `Create` button is still too closely attached to text box.
2. Missing of the `Create a new section...` text and a pop-up suggesting what the name could be
3. `Or select one available section to add users` is too close to `create section` text box
4. Nip-pick, The Mockup calls for text saying `input the name of the new section` and PR text says 'New Section Name`.